pep-0678.rst

```diff
@@ -229,6 +232,49 @@ proposed ``__note__`` semantics, but this would be rarely and inconsistently
 applicable.
 
 
+Augment the ``raise`` statement
+-------------------------------
+One discussion proposed ``raise Exception() with "note contents"`` .
+This would be a strictly additional feature over adding the attribute and modifying
+the traceback-formatting code, and we do not expect ``__note__`` to be so
+frequently-used as to justify new syntax.
+
+
+Structured API rather than string attribute
+-------------------------------------------
+Instead of a string-or-None, ``__note__`` could be a new object with a structured API
+and a ``__str__()`` method for printing at trackback time.
+We have not identified any scenario where libraries would want to do anything but either
+concatenate or replace notes, and so the additional complexity does not seem justified.
+
+Permitting ``__note__`` to be "None or any object with a ``__str__()`` method" would
+allow for experimentation with such proposals, and forward-compatibility one was adopted.
+We see this as a flexibility/usability tradeoff, and in the absence of any concrete
+use-case we narrowly prefer to guarantee that notes are strings but remain open to
+revisiting the decision.
+
+
+Add a helper function ``contexlib.add_exc_note()``
+--------------------------------------------------
+We don't expect notes to be used frequently enough to justify this either, but
+provide the following recipe::
+
+    @contextlib.contextmanager
+    def add_exc_note(
+        note: str,
+        exc: type[BaseException] | tuple[type[BaseException], ...] = BaseException,
+        concatenate: str | None = "\n\n",
+    ):
+        try:
+            yield
+        except exc as err:
+            if concatenate is None or err.__note__ is None:
+                err.__note__ = note
+            else:
+                err.__note__ = err.__note__ + concatenate + note
+            raise
```

Comment:
Maybe you can provide a simple example of how this would be used? I'd also trim down the recipe down significantly -- people who want to constrain the exception or tweak the concatenation behavior can extend the recipe.